text/src/autogluon/text/text_prediction/predictor/predictor.py

```diff
@@ -0,0 +1,398 @@
+
+        Returns
+        -------
+        ret : a single number or a dict of metric --> metric scores
+            Output
+        """
+        return self._model.evaluate(data, metrics=metrics)
+
+    def predict(self, dataset, as_pandas=False):
+        """Get the prediction from
+
+        Returns
+        -------
+        output
+            Array of predictions. One element corresponds to the prediction value of one
+        as_pandas
+            Whether to convert the output to a pandas dataframe
+        """
+        assert self._model is not None, 'Model does not seem to have been constructed. Have you called fit(), or load()?'
+        output = self._model.predict(dataset)
+        if as_pandas:
+            output = pd.DataFrame({self.label: output})[self.label]
+        return output
+
+    def predict_proba(self, dataset, as_pandas=False):
+        """Predict the probability from the input
+
+        Parameters
+        ----------
+        dataset
+            The dataset
+        as_pandas
+            Whether to convert the output to pandas dataframe
+
+        Returns
+        -------
+        output
+            The output matrix
+        """
+        assert self._model is not None,\
+            'Model does not seem to have been constructed. ' \
+            'Have you called fit(), or load()?'
+        output = self._model.predict_proba(dataset)
+        if as_pandas:
+            output = pd.DataFrame(output)
+        return output
+
+    def extract_embedding(self, dataset, stochastic_chunk=None, num_repeat=None, as_pandas=False):
+        """Extract the feature from the neural network
+
+        Returns
+        -------
+        output
+        """
+        assert self._model is not None, 'Model does not seem to have been constructed. ' \
+                                        'Have you called fit(), or load()?'
+        output = self._model.extract_embedding(dataset,
+                                               stochastic_chunk=stochastic_chunk,
+                                               num_repeat=num_repeat)
+        if as_pandas:
+            output = pd.DataFrame({self.label: output})
+        return output
+
+    def save(self, dir_path):
+        """Save the model to directory path
+
+        The model will be saved to directory path:
+
+
+        Parameters
+        ----------
+        dir_path
+            The directory path to save the model artifacts
+
+        """
+        assert self._model is not None, 'Model does not seem to have been constructed.' \
+                                        ' Have you called fit(), or load()?'
+        os.makedirs(dir_path, exist_ok=True)
+        with open(os.path.join(dir_path, 'text_predictor_assets.json'), 'w') as of:
+            json.dump({'backend': self._backend,
+                       'label': self._label}, of)
+        self._model.save(os.path.join(dir_path, 'saved_model'))
```

Comment:
set `self._model` to None and then save `TextPredictor`. 
For loading, make sure to load `self._model after loading `TextPredictor`.

Reply:
Or is there nothing to save in TextPredictor besides `text_predictor_assets`?

Reply:
Yes, there's nothing to save besides those in assets.